### ISSUE TITLE
Warping from Nintendo Presents screen loads incorrectly

### ISSUE THREAD

**Issue by cbrgray** [Arnethegreat/yoshisisland-practice-hack#9]
Loading directly into the start of 1-2 from game boot and then finishing the level normally spawns corrupted tile collision data that prevents the end yoshi from leaving, causing a soft lock. This tile data appears to come from screen region $7D, just to the left of the expected $7E/$7F.

Not sure if other levels have the same behaviour.

**Comment by cbrgray** [comment]
### TL;DR warping before selecting a game file skips uploading goal ring fanfare sound data into the APU which can break the game.


Appears to be a buffer overflow of the sound queue (`$7E0059`) triggered on entering the goal ring:
* goal ring sounds @ `$02A7B5` loaded into the sound queue
* after this point, the sound queue never gets processed/emptied (APU hang?), and instead continues increasing far above its intended capacity (`$0085D8`)
* the overflow corrupts `$7E0073` (a value which is normally either 0 or 2 and is used to index into `$109085`) to a much higher value
* causes a read of arbitrary garbage @ `$109091`
* this is then used by the tilemap column loader to determine which screen region to load from

The sound queue is normally processed and decremented @ `$00C058/$7EC058`
I think the goal ring fanfare isn't part of the level sound data and only gets loaded into the APU in gamemode $20 (overworld load), which would be skipped by warping early. When the APU tries to play the fanfare, it either plays garbage or hangs, causing the CPU bug.

Gamemode $20 loads:
- default music track `$1F82E6`
- intro, map, castle/fort samples `$1EEE5A`
- overworld music track `$1EC122`


This bug is also triggered by the sounds that play during the death cutscene e.g. when falling into a pit.
